src/domain/wallets/index.types.d.ts

```diff
@@ -12,11 +10,47 @@ type Deprecated = {
   readonly feeUsd: number
 }
 
+type InitiationViaIntraledger = {
+  readonly type: PaymentInitiationMethod["IntraLedger"]
+  readonly walletId: WalletId
+  readonly counterPartyUsername: Username
+}
+
+type InitiationViaLn = {
+  readonly type: PaymentInitiationMethod["Lightning"]
+  readonly paymentHash: PaymentHash
+  readonly pubkey: Pubkey
+}
+
+type InitiationViaOnChain = {
+  readonly type: PaymentInitiationMethod["OnChain"]
+  readonly address: OnChainAddress
+}
+
+type InitiationViaOnChainLegacy = {
+  readonly type: PaymentInitiationMethod["OnChain"]
+  readonly address?: OnChainAddress
+}
+
+type SettlementViaIntraledger = {
+  readonly type: SettlementMethod["IntraLedger"]
+  readonly walletId: WalletId
+  readonly counterPartyUsername: Username | null
+}
+
+type SettlementViaLn = {
+  readonly type: SettlementMethod["Lightning"]
+  paymentSecret: PaymentSecret | null
```

Comment:
this shouldn't be `null`? we should always know the `paymentSecret`  once this has been settled? maybe the reason this may not be `null` is if the payment is in flight?

Reply:
This was a temporary thing because we don't persist this yet for successful payments. I'm planning to add this in after the persist-payments PR (#665) is merged.